kraken-build/src/kraken/std/python/tasks/publish_task.py

```diff
@@ -33,31 +33,28 @@ def get_relationships(self) -> Iterable[TaskRelationship]:
 
     def execute(self) -> TaskStatus:
         credentials = self.index_credentials.get()
-        repository_url = self.index_upload_url.get().rstrip("/") + "/"
         command = [
-            *self.twine_cmd.get(),
-            "upload",
-            "--repository-url",
-            repository_url,
-            "--verbose",
+            "uv",
+            "publish",
+            "--publish-url",
+            self.index_upload_url.get(),
             *[str(x.absolute()) for x in self.distributions.get()],
         ]
-        if credentials:
-            command += [
-                "--username",
-                credentials[0],
-                "--password",
-                credentials[1],
-            ]
-        if not self.interactive.get():
-            command.append("--non-interactive")
         if self.skip_existing.get():
-            command.append("--skip-existing")
+            command.extend(["--check-url", self.index_check_url.get() or self.index_index_url.get()])
+
+        env = os.environ.copy()
+        if credentials:
+            # See https://docs.astral.sh/uv/guides/package/#publishing-your-package
+            # NOTE: PyPI does not support publishing with username and password anymore,
+            #       should we log a warning if the username is not __token__?
```

Comment:
I think that's reasonable! The check could be `"pypi.org/" in self.index_upload_url.get() and credentials[0] != "__token__"`. Would be good to include a bit of info as well like `logger.warning("since 2025-MM-DD, PyPi no longer allows publishing with username/password, see https://...")`